game/src/ecs/systems/InteractionSystem.java

```diff
@@ -0,0 +1,58 @@
+package ecs.systems;
+
+import ecs.components.Component;
+import ecs.components.InteractionComponent;
+import ecs.components.MissingComponentException;
+import ecs.components.PositionComponent;
+import ecs.components.ai.AITools;
+import ecs.entities.Entity;
+import ecs.entities.Hero;
+import java.util.Optional;
+
+import mydungeon.ECS;
+
+public class InteractionSystem {
```

Comment:
fehlt hier nicht `extends ECS_System`?

Reply:
habe das mal auf Funktionaler umgebaut bin aber sicher es geht noch schöner

Reply:
InteractionSystem ist vielleicht nicht die beste Wortwahl, vielleicht InteractionUtils. 

Also was es macht ist nichts anderes, als eine Utility Methode zur Verfügung zu stellen, welche weder wirklich in das KeyboardSystem passt noch wirklich in die Komponente. Da es ein Suchen der Komponente ist, welche am nächsten zum Helden ist.

Reply:
Ok ich verstehe. 
Dann nenn das Ding auf keinen Fall System und schieb es aus dem Package raus. 

Idee: Leg eine Tool-Klasse an für sowas an. Im Keyboard-System kannst du dann ja schauen, wenn der Spieler den Knopf drückt, ob und welche Interagierbare Entität in der nähe ist.